Blog: Demystifying session recording – examining the (trivial) performance hit (#4098)
PostHog/posthog.com: contents/blog/session-recording-performance.md

---
date: 2022-08-31

> title: "Session Recording Demystified: Examining the (trivial) performance hit"

Author: Possible alternative headlines:

- Will session recording slow down your website? Examining the (trivial) impact

> title: "Session Recording Demystified: Examining the (trivial) performance hit"

Author: ```suggestion
title: "Benchmarking the (trivial) impact of session recording on performance"
```

> **Verdict:** MutationObserver has **no** discernible impact to page memory.

### Network: Transmission

Network is the bigger metric to interrogate. After all, computers have gotten faster, memory has gotten cheaper, but bad internet is still… pretty damn common. How badly does SessionRecording impact your internet speed? 

Let’s take a look at the Network tab on all three apps. 

**Network Tab: Brevy**

![Average Packet Size: 310 bytes](../images/blog/session-recording-performance/Screen_Shot_2022-08-31_at_1.45.10_AM.png)

Average Packet Size: 310 bytes

**Network Tab: Commandbar:** 

![Average Packet Size: 309 bytes](../images/blog/session-recording-performance/Screen_Shot_2022-08-30_at_11.14.03_PM.png)

Average Packet Size: 309 bytes

**Network Tab: Explo:** 

![Average Packet Size: 309 bytes](../images/blog/session-recording-performance/Screen_Shot_2022-08-31_at_12.12.48_AM.png)

Average Packet Size: 309 bytes

All of the gzip-compressed transmissions? That’s **rrweb** / **PostHog** relaying the mutation data to a PostHog host for storage. 

You will notice two things. 

> - These packets are **tiny** in comparison to anything else—300 or so bytes. Bytes, not kilobytes. That’s it!

Reviewer: Of note, the 309 bytes is the size of the response payload, not the size of the uploaded data (which is significantly larger)

Author: Thanks Rick, I'll get a PR up to update the live article. Apologies for the rush! Damn newsletters!
